src/components/layouts/FilterBar.tsx

```diff
@@ -14,19 +19,57 @@ import { IAssigneeCombined, View } from '@/types/interfaces'
 import { CrossIcon, FilterByAsigneeIcon } from '@/icons'
 import { ViewModeSelector } from '../inputs/ViewModeSelector'
 import { FilterByAssigneeBtn } from '../buttons/FilterByAssigneeBtn'
+import { Token } from '@/types/common'
+import FilterButtonGroup from '@/components/buttonsGroup/FilterButtonsGroup'
 
-export const FilterBar = ({ updateViewModeSetting }: { updateViewModeSetting: (mode: View) => void }) => {
+export const FilterBar = ({
+  updateViewModeSetting,
+  getTokenPayload,
+}: {
+  updateViewModeSetting: (mode: View) => void
+  getTokenPayload: () => Promise<Token>
+}) => {
   const [searchText, setSearchText] = useState('')
-
+  const [activeButtonIndex, setActiveButtonIndex] = useState<number>()
   const { view } = useSelector(selectTaskBoard)
-
   const { assignee } = useSelector(selectTaskBoard)
 
   const { renderingItem: _assigneeValue, updateRenderingItem: updateAssigneeValue } = useHandleSelectorComponent({
     item: assignee[0],
     type: SelectorType.ASSIGNEE_SELECTOR,
   })
 
+  const filterButtons = [
+    {
+      name: 'My tasks',
+      onClick: async (index: number) => {
+        const payload = await getTokenPayload()
+        store.dispatch(setFilteredAsignee({ id: payload.internalUserId }))
```

Comment:
It would be much cleaner and easier to understand if we passed the internalUserId as prop to this component directly.

Use `getTokenPayload` in the Page server component and pass `internalUserId` as prop instead.

Reply:
Lets store the `payload` in the redux store rather than passing as a prop to the component it could be required to use the `payload` in other places in the future.